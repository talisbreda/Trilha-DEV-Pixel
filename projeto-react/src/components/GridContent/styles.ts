import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface Props {
  theme: Theme;
}

export const Container = styled.div<Props>`
  ${({ theme }) => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const Html = styled.div<Props>`
  ${({ theme }) => css`
    margin: ${theme.spacings.huge} 0;
  `}
`;
