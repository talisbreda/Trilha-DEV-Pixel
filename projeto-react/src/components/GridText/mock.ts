export type GridTextElement = {
  title: string;
  description: string;
};

export type GridTextMock = {
  background: boolean;
  title: string;
  description: string;
  grid: Array<GridTextElement>;
};

export const mock: GridTextMock = {
  background: false,
  title: 'My grid',
  description:
    'Aliquam ac quam posuere, sagittis erat sit amet, eleifend felis. Sed ullamcorper ornare gravida.',
  grid: [
    {
      title: 'Teste 1',
      description: `Donec tincidunt tincidunt massa ac volutpat. Quisque vitae diam pretium
        porta purus nec, vulputate mauris. Vivamus vitae dapibus tortor. Ut eu gravida
        felis. In dictum dolor consequat, porta ante nec, viverra magna. Morbi nec erat eros.
        In vitae ipsum eget odio congue tristique non quis odio.'`,
    },
    {
      title: 'Teste 2',
      description: `Donec tincidunt tincidunt massa ac volutpat. Quisque vitae diam pretium
        porta purus nec, vulputate mauris. Vivamus vitae dapibus tortor. Ut eu gravida
        felis. In dictum dolor consequat, porta ante nec, viverra magna. Morbi nec erat eros.
        In vitae ipsum eget odio congue tristique non quis odio.'`,
    },
    {
      title: 'Teste 3',
      description: `Donec tincidunt tincidunt massa ac volutpat. Quisque vitae diam pretium
        porta purus nec, vulputate mauris. Vivamus vitae dapibus tortor. Ut eu gravida
        felis. In dictum dolor consequat, porta ante nec, viverra magna. Morbi nec erat eros.
        In vitae ipsum eget odio congue tristique non quis odio.'`,
    },
  ],
};
