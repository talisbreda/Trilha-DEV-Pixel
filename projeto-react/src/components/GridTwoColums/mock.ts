export default {
  title: 'Grid two columns',
  text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Autem itaque ullam, minus, non dolorem vitae atque doloremque assumenda voluptatem illum aliquid obcaecati voluptas omnis, maxime a culpa ea iusto ab!',
  imageSrc: 'assets/images/javascript.svg',
  background: false,
};
