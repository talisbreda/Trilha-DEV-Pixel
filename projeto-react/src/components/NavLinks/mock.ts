export default [
  {
    children: 'Link 1',
    link: '#target1',
    newTab: false,
  },
  {
    children: 'Link 2',
    link: '#target2',
    newTab: true,
  },
  {
    children: 'Link 3',
    link: '#target3',
    newTab: true,
  },
  {
    children: 'Link 4',
    link: '#target4',
    newTab: true,
  },
  {
    children: 'Link 5',
    link: '#target5',
    newTab: false,
  },
  {
    children: 'Link 6',
    link: '#target6',
    newTab: false,
  },
  {
    children: 'Link 7',
    link: '#target7',
    newTab: false,
  },
  {
    children: 'Link 8',
    link: '#target8',
    newTab: false,
  },
  {
    children: 'Link 9',
    link: '#target9',
    newTab: false,
  },
  {
    children: 'Link 0',
    link: '#target0',
    newTab: false,
  },
  {
    children: 'Link 1',
    link: '#target11',
    newTab: false,
  },
  {
    children: 'Link 2',
    link: '#target12',
    newTab: false,
  },
  {
    children: 'Link 3',
    link: '#target3',
    newTab: false,
  },
];
