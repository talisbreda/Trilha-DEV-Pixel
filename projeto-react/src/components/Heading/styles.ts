import styled, { css, FlattenSimpleInterpolation } from 'styled-components';
import { Theme } from '../../styles/theme';

interface Props {
  theme: Theme;
  darkColor: boolean;
  as: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  size: 'huge' | 'large' | 'medium' | 'small';
  upperCase: boolean;
}

type sizes = {
  small: (theme: Theme) => FlattenSimpleInterpolation;
  medium: (theme: Theme) => FlattenSimpleInterpolation;
  large: (theme: Theme) => FlattenSimpleInterpolation;
  huge: (theme: Theme) => FlattenSimpleInterpolation;
};

const mediaFont = (theme: Theme) => css`
  @media ${theme.media.ltMedium} {
    font-size: ${theme.font.sizes.large}
  }
`;

const titleSize: sizes = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  large: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    ${mediaFont(theme)};
  `,
};

const titleCase = (upperCase: boolean) => css`
  text-transform: ${upperCase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<Props>`
  ${({ theme, darkColor, size, upperCase }) => {
    return css`
        color: ${darkColor ? theme.colors.primaryColor : theme.colors.white};
        ${titleSize[size](theme)};
        ${titleCase(upperCase)};
    `;
  }}
`;
