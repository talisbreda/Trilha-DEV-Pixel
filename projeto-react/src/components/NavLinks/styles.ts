import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface Props {
  theme: Theme;
}

export const Container = styled.nav<Props>`
  ${({ theme }) => css`
    background: #fff;
    display: flex;
    flex-flow: row wrap;

    @media ${theme.media.ltMedium} {
      flex-flow: column nowrap;
      align-items: center;
    }
  `}
`;
