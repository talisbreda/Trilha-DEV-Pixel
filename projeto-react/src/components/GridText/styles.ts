import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Theme } from '../../styles/theme';
import { Title as HeadingContainer } from '../Heading/styles';

interface Props {
  theme: Theme;
}

export const Container = styled.div<Props>`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div<Props>`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large}
  `}
`;

export const GridElement = styled.div<Props>`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -3rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
