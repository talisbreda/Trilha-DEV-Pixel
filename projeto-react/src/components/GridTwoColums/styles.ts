import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';
import { Title } from '../Heading/styles';

interface Props {
  theme: Theme;
}

export const Container = styled.div<Props>`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.ltMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      margin-bottom: ${theme.spacings.huge};
    }
  `}
`;

export const TextContainer = styled.div<Props>`
  ${({ theme }) => css`
    @media ${theme.media.ltMedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div<Props>`
  ${() => css``}
`;

export const Image = styled.img<Props>`
  ${() => css``}
`;
