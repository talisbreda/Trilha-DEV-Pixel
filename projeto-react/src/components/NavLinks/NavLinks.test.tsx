import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { NavLinks } from '.';
import mock from './mock';

describe('<NavLinks />', () => {
  it('should render', () => {
    renderTheme(<NavLinks links={mock} />);
    expect(screen.getAllByRole('link')).toHaveLength(mock.length);
  });

  it('should not render links', () => {
    renderTheme(<NavLinks />);
    expect(screen.getAllByText(/link/i)).toHaveLength(mock.length);
  });

  it('should not match snapshot', () => {
    renderTheme(<NavLinks links={mock} />);
    expect(screen.getAllByRole('link')).toMatchSnapshot();
  });
});
