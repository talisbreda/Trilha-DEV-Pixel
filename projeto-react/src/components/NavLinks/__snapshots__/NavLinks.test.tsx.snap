// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> should not match snapshot 1`] = `
Array [
  <a
    class="sc-bcXHqe dfnVxD"
    href="#target1"
    target="_self"
  >
    Link 1
  </a>,
  <a
    class="sc-bcXHqe dfnVxD"
    href="#target2"
    target="_blank"
  >
    Link 2
  </a>,
  <a
    class="sc-bcXHqe dfnVxD"
    href="#target3"
    target="_blank"
  >
    Link 3
  </a>,
  <a
    class="sc-bcXHqe dfnVxD"
    href="#target4"
    target="_blank"
  >
    Link 4
  </a>,
  <a
    class="sc-bcXHqe dfnVxD"
    href="#target5"
    target="_self"
  >
    Link 5
  </a>,
  <a
    class="sc-bcXHqe dfnVxD"
    href="#target6"
    target="_self"
  >
    Link 6
  </a>,
  <a
    class="sc-bcXHqe dfnVxD"
    href="#target7"
    target="_self"
  >
    Link 7
  </a>,
  <a
    class="sc-bcXHqe dfnVxD"
    href="#target8"
    target="_self"
  >
    Link 8
  </a>,
  <a
    class="sc-bcXHqe dfnVxD"
    href="#target9"
    target="_self"
  >
    Link 9
  </a>,
  <a
    class="sc-bcXHqe dfnVxD"
    href="#target0"
    target="_self"
  >
    Link 0
  </a>,
  <a
    class="sc-bcXHqe dfnVxD"
    href="#target11"
    target="_self"
  >
    Link 1
  </a>,
  <a
    class="sc-bcXHqe dfnVxD"
    href="#target12"
    target="_self"
  >
    Link 2
  </a>,
  <a
    class="sc-bcXHqe dfnVxD"
    href="#target3"
    target="_self"
  >
    Link 3
  </a>,
]
`;
